txscript/engine.go

```diff
@@ -900,17 +899,14 @@ func NewEngine(scriptPubKey []byte, tx *wire.MsgTx, txIdx int, flags ScriptFlags
 
 			witProgram = scriptPubKey
 		case len(tx.TxIn[txIdx].Witness) != 0 && vm.bip16:
-			pops, err := parseScript(scriptSig)
-			if err != nil {
-				return nil, err
-			}
-
 			// The sigScript MUST be *exactly* a single canonical
 			// data push of the witness program, otherwise we
 			// reintroduce malleability.
-			if len(pops) == 1 && canonicalPush(pops[0]) &&
-				IsWitnessProgram(pops[0].data) {
-				witProgram = pops[0].data
+			dataPush := vm.scripts[0][0]
```

Comment:
This will crash if there are no pushes in the signature script.  Maybe the following?

```Go
sigPops := vm.scripts[0]
if len(sigPops) == 1 && canonicalPush(sigPops[0]) &&
	IsWitnessProgram(sigPops[0].data) {
```